rct_optimizations/test/include/rct_optimizations_tests/pose_generator.h

```diff
@@ -25,70 +26,258 @@ struct PoseGenerator
    * @param target_origin The position of the target
    * @return A vector of camera positions & orientations
    */
-  virtual std::vector<Eigen::Isometry3d> generate(const Eigen::Vector3d &target_origin) const = 0;
+  virtual std::vector<Eigen::Isometry3d> generate(const Eigen::Isometry3d &target_origin) = 0;
+
+  virtual double getZRot() = 0;
 };
 
 /**
  * @brief Generates camera poses in a hemisphere pattern
  */
 struct HemispherePoseGenerator : PoseGenerator
 {
-  inline HemispherePoseGenerator(double r_ = 2.0,
-                                 unsigned theta_cnt_ = 10,
-                                 unsigned phi_cnt_ = 10)
+protected:
+  inline HemispherePoseGenerator(double r_,
+                                 unsigned theta_cnt_,
+                                 unsigned phi_cnt_,
+                                 double z_rot_min_,
+                                 double z_rot_max_,
+                                 const Eigen::Isometry3d& target_offset_,
+                                 unsigned long seed_ = std::mt19937::default_seed)
     : r(r_)
     , theta_cnt(theta_cnt_)
     , phi_cnt(phi_cnt_)
+    , target_offset(target_offset_)
+    , mt_rand(seed_)
+    , z_sampler(z_rot_min_, z_rot_max_)
+  {
+
+  }
+
+public:
+  inline HemispherePoseGenerator(double r_,
+                                 unsigned theta_cnt_,
+                                 unsigned phi_cnt_,
+                                 double z_rot_,
+                                 const Eigen::Isometry3d& target_offset_)
+    : HemispherePoseGenerator(r_, theta_cnt_, phi_cnt_, z_rot_, z_rot_, target_offset_)
```

Comment:
Is this intended to provide a random Z rotation on [`-z_rot`, `z_rot`], or just to lock the Z-rotation at a specific angle?

Reply:
This is intended to lock the Z-rotation to the angle provided in `z_rot_`.